modules/frontend/handler.go

```diff
@@ -67,6 +68,17 @@ func (f *handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	resp, err := f.roundTripper.RoundTrip(r)
 	if err != nil {
 		writeError(w, err)
+		if err == context.Canceled {
+			level.Info(f.logger).Log(
+				"tenant", orgID,
+				"method", r.Method,
+				"traceID", traceID,
+				"url", r.URL.RequestURI(),
+				"duration", time.Since(start).String(),
+				"response_size", 0,
+				"status", StatusClientClosedRequest,
```

Comment:
is this necessarily true? if the query-frontend itself cancels the context won't it return the same error?

Reply:
thank you for the review. I agree, is there a way to differentiate between client and query-frontend cancellation? Will this work`if err == ctx.Err()`?

Reply:
ok, apologies. I didn't look closely at this PR/code before I responded. 

Let's do this:

- Instead of only logging an error if context is canceled, let's log every error. 
- For status let's just use a generic 500 since we don't know what the cause of the error is. 
- Add this log line to both the `resp == nil` and `err != nil` blocks
- change `writeError` to return an error instead of calling `server.WriteError` directly. That way we can also add the error string. 
- Log a new field after "status": 
  `"err", err.String()`

Reply:
thank you, I've made changes.